packages/runed/src/lib/utilities/useGeolocation/useGeolocation.svelte.ts

```diff
@@ -0,0 +1,95 @@
+import { IsSupported } from "../IsSupported/IsSupported.svelte.js";
+
+type UseGeolocationOptions = Partial<PositionOptions> & {
+	/**
+	 * Whether to start the watcher immediately upon creation. If set to `false`, the watcher
+	 * will only start tracking the position when `resume()` is called.
+	 *
+	 * @defaultValue true
+	 */
+	immediate?: boolean;
+};
+
+type WritableProperties<T> = {
+	-readonly [P in keyof T]: T[P];
+};
+
+/**
+ * Reactive access to the browser's [Geolocation API](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API).
+ */
+export function useGeolocation(options: UseGeolocationOptions = {}) {
+	const {
+		enableHighAccuracy = true,
+		maximumAge = 30000,
+		timeout = 27000,
+		immediate = true,
+	} = options;
+
+	const isSupported = new IsSupported(() => navigator && "geolocation" in navigator);
+
+	let locatedAt = $state<number | null>(null);
+	let error = $state.raw<GeolocationPositionError | null>(null);
+	let coords = $state<WritableProperties<Omit<GeolocationPosition["coords"], "toJSON">>>({
+		accuracy: 0,
+		latitude: Number.POSITIVE_INFINITY,
+		longitude: Number.POSITIVE_INFINITY,
+		altitude: null,
+		altitudeAccuracy: null,
+		heading: null,
+		speed: null,
+	});
+	let isPaused = $state(false);
+
+	function updatePosition(position: GeolocationPosition) {
+		locatedAt = position.timestamp;
+		coords.accuracy = position.coords.accuracy;
```

Comment:
I'm not sure if this makes sense here or if we use `$state.raw` for the coords. This feels more fine-grained but open to feedback.